src/main/scala/PrivateHome/data.scala

```diff
@@ -1,47 +1,180 @@
 package PrivateHome
 
+import PrivateHome.Devices.MHz.mhzSwitch
+import PrivateHome.Devices.MQTT.mqttSwitch
 import PrivateHome.Devices.Switch
+import scalikejdbc._
 
-import scala.xml._
 
 object data {
+  /**
+   * An map containing all settings
+   */
+  val settings: Map[String, Int] = Map(("port", 2888), ("ip", 565))
+
+
+  GlobalSettings.loggingSQLAndTime = LoggingSQLAndTimeSettings(
+    enabled = false,
+    singleLineMode = false,
+    printUnprocessedStackTrace = false,
+    stackTraceDepth = 15,
+    logLevel = Symbol("debug"),
+    warningEnabled = true,
+    warningThresholdMillis = 3000L,
+    warningLogLevel = Symbol("warn")
+  )
+
+
+  Class.forName("org.h2.Driver")
+  ConnectionPool.singleton("jdbc:h2:./daten/Devices", "user", "pass")
+
+  implicit val session: AutoSession.type = AutoSession
+  var devices: Map[String, Switch] = Map()
+  var mhzId: Map[String, String] = Map()
 
   /**
-   * This variable returns a List containing all ids.
+   * Generates the Table strucktures for the Database
    */
-  val IDs: List[String] = {
-    val xml = XML.load("src/main/scala/PrivateHome/id.xml")
-    val prep: NodeSeq = xml \\ "id"
-    val result = for (i <- 0 until prep.length) yield prep(i).text
-    result.toList
+  def create(): Unit = {
+
+    // Drops old Tables to Delet Data and make it possible to regenerate them
+    sql"""DROP TABLE IF EXISTS `Mhz`;
+         DROP TABLE IF EXISTS `Devices`;""".execute().apply()
+
+
+    val devices =
+      sql"""
+                    CREATE TABLE `Devices` (
+                    `id` varchar(5) NOT NULL,
+                    `name` varchar(64) NOT NULL,
+                    `type` varchar(16) NOT NULL,
+                    `state` decimal(5,4) NOT NULL,
+                    `keepState` boolean NOT NULL,
+                    PRIMARY KEY (`id`))
+         """.execute.apply()
+
+
+    sql"""
+           CREATE TABLE `Mhz` (
+           `id` varchar(5) NOT NULL,
+           `systemcode` varchar(5) NOT NULL,
+           `unitcode` varchar(5) NOT NULL,
+           PRIMARY KEY (`id`),
+           CONSTRAINT `Mhz_ibfk_1` FOREIGN KEY (`id`) REFERENCES `Devices` (`id`))
+         """.execute().apply()
+
+
+    // insert initial data
```

Comment:
Did you forget MQTT? Yes you did!

Reply:
NO because MQTT doesn't need any Switch dependent settings outer than the normal Switch data look at Switch.scala und mqttswitch.scala only variables are also defined in Switch

Reply:
ok